src/vos/vos_io.c

```diff
@@ -410,37 +447,16 @@ akey_fetch_recx(daos_handle_t toh, const daos_epoch_range_t *epr,
 			rsize = ent_array.ea_inob;
 		D_ASSERT(rsize == ent_array.ea_inob);
 
-		if (dcb_is_valid(csum) &&
-		    csum_copied * ent->en_csum.cs_len < csum->cs_buf_len &&
-		    csum->cs_chunksize > 0) {
-			uint8_t	    *csum_ptr;
-			daos_size_t  csum_nr;
-
-			D_ASSERT(lo >= recx->rx_idx);
-			/** Make sure the entity csum matches expected csum */
-			D_ASSERT(csum->cs_len == ent->en_csum.cs_len);
-			D_ASSERT(csum->cs_type == ent->en_csum.cs_type);
-			D_ASSERT(csum->cs_chunksize ==
-				 ent->en_csum.cs_chunksize);
-
-			/** Note: only need checksums for requested data, not
-			 * necessarily all checksums from the entire recx entry
-			 */
-			csum_ptr = dcb_off2csum(csum,
-				(uint32_t)((lo - recx->rx_idx) * rsize));
-			csum_nr = csum_chunk_count(csum->cs_chunksize,
-						   lo, hi, rsize);
-			memcpy(csum_ptr, ent->en_csum.cs_csum,
-			       csum_nr * ent->en_csum.cs_len);
-			if (DAOS_FAIL_CHECK(DAOS_CHECKSUM_FETCH_FAIL))
-				csum_ptr[0] += 2; /* poison the checksum */
-
-			csum_copied += csum_nr;
-			D_ASSERT(csum_copied <= csum->cs_nr);
-		}
+		bio_iov_set(&biov, ent->en_addr, nr * ent_array.ea_inob);
+
+		rc = bsgl_dcb_resize(ioc);
```

Comment:
why don't we just do the resize within vic_update_biov?  In addition, do we want to resize even if checksum is disabled?